### ISSUE TITLE
docker-compose pull argus fails 

### ISSUE THREAD

**Issue by jrhbcn** [release-argus/Argus#57]
**Describe the bug**
Trying to `docker-compose pull` argus using the provided docker-compose documentation fails with

`failed to register layer: ApplyLayer exit status 1 stdout:  stderr: lchown /argus: invalid argument`

**To Reproduce**
1. Fill docker-compose with recommended settings
3. Try to docker-compose pull argus

**Expected behavior**
argus pulls, and I am able to `docker-compose up -d` it. 

**Desktop (please complete the following information):**
I am running docker 20.10.12 in a LXC container under proxmox. I have running several other docker apps (vaultwarden, uptime-kuma, dashy, paperless-ngx, filebrowser, etc.) in the same docker-compose file and all of them are working.

**Comment by JosephKav** [comment]
Hmm, very odd. Seems to be a known issue in Docker though
> docker: failed to register layer: Error processing tar file(exit status 1): lchown <FILE>: invalid argument

> This error occurs when the number of available entries in /etc/subuid or /etc/subgid is not sufficient. The number of entries required vary across images. However, 65,536 entries are sufficient for most images

https://docs.docker.com/engine/security/rootless/#docker-pull-errors

> /etc/subuid and /etc/subgid should contain at least 65,536 subordinate UIDs/GIDs for the user. In the following example, the user testuser has 65,536 subordinate UIDs/GIDs (231072-296607).

https://docs.docker.com/engine/security/rootless/#prerequisites

Could you try
> grep ^DOCKER_USER: /etc/subuid
> grep ^DOCKER_GROUP: /etc/subgid

And check that they're both 65536+? If they are, then try adding any new container and see if they all have this issue

**Comment by jrhbcn** [comment]
Hi @JosephKav,

Many thanks for looking into this. First, I confirmed that the main problem is using unprivileged LXC containers in proxmox to run docker. Just to test, I have created a privileged one on my proxmox server and then the argus container indeed works fine there. 

About trying to solve running argus in the unprivileged one, it seems a bit more complicated. I followed the recommended [guide](https://pve.proxmox.com/wiki/Unprivileged_LXC_containers) in proxmox to give access to container users to mount disks (in the guide user 1005). And as I said, all other containers I have used work.

I could not find a proper solution in your recommendations/links, however, I also had to include a mapping for group "_apt" to make "apt update/upgrade" work in the container (following this other [post](https://forum.proxmox.com/threads/lxc-permission-issues-apt-fails.64387/)). Would it be possible that the argus container is using a group/user that also need to be mapped between host and LXC?

Regards,

**Comment by JosephKav** [comment]
Did you setup any user/group bindings to get Docker working on those other containers? `vaultwarden` and `uptime-kuma` look like they run as `root` inside the container and I don't think I should copy that as that sounds like it'd be less secure.
The container runs as `nobody`, so you could try mapping that user/group?
```bash
❯ id nobody
uid=65534(nobody) gid=65534(nogroup) groups=65534(nogroup)
```

> I could not find a proper solution in your recommendations/links, however, I also had to include a mapping for group "_apt" to make "apt update/upgrade" work in the container (following this other [post](https://forum.proxmox.com/threads/lxc-permission-issues-apt-fails.64387/)).

Did you exec to the container and try and `apt update/upgrade`, or try and build it locally on that unprivileged LXC container?

**Comment by jrhbcn** [comment]
Thanks @JosephKav, it is now fixed. Adding the nobody user to the list of mapped user/group IDs did the trick. 

For reference in case anybody else stumbles on the same problem was in my proxmox lxc configuration. I just mapped 65530 users, so I was missing 5 ;) just making sure that all 65535 users are mapped and everything works again.

About your comment on using root. This is one of the advantages of running docker on unprivileged LXC ones, root user in docker containers is mapped to a normal user in the host so it is as secure as the nobody user.

Thanks again for your help, I will start using argus so expect some more questions or suggestions from my part ;)
